bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/RefreshingOAuth2CredentialsInterceptor.java

```diff
@@ -195,19 +204,17 @@ private HeaderCacheElement getCachedHeader() throws IOException {
   /**
    * Perform a credentials refresh.
    */
-  private void doRefresh() {
-    boolean requiresRefresh = false;
+  @VisibleForTesting
+  boolean doRefresh() {
+    boolean performRefresh = false;
     synchronized (isRefreshing) {
-      if (!isRefreshing.get()) {
+      if (!isRefreshing.get() && getCacheState(headerCache.get()) != CacheState.Good) {
         isRefreshing.set(true);
-        requiresRefresh = true;
-        HeaderCacheElement headerCache = this.headerCache.get();
-        CacheState state = (headerCache == null) ? CacheState.Expired : headerCache.getCacheState();
-        requiresRefresh = state != CacheState.Good;
+        performRefresh = true;
       }
     }
-    if (!requiresRefresh) {
-      return;
+    if (!performRefresh) {
```

Comment:
It's not refreshed at this point.  Does doRefresh work?  Maybe requiresRefresh like what was there before?